src/bots/HighSellVolumeBot.js

```diff
@@ -0,0 +1,255 @@
+        const buyToken = market.tokenB
+        this._doRoutineLiquidityCheck(sellToken, buyToken)
+      })
+    }, this._checkTimeInMilliseconds)
+  }
+
+  async _doStop () {
+    logger.debug({ msg: 'Bot stopped: ' + this.name })
+  }
+
+  async _doRoutineLiquidityCheck (sellToken, buyToken) {
+    // Do ensure liquidity on the market
+    auctionLogger.debug({
+      sellToken,
+      buyToken,
+      msg: "Doing a routine check. Let's see if we will be able to ensure the liquidity"
+    })
+    return this._checkBuyLiquidity({
+      sellToken,
+      buyToken,
+      from: this._botAddress
+    })
+  }
+
+  async _checkBuyLiquidity ({ sellToken, buyToken, from }) {
+    this._lastCheck = new Date()
+    let liquidityWasChecked
+    try {
+      // We must check both auction sides
+      await this._checkEnoughBalance({ sellToken, buyToken, from })
+      await this._checkEnoughBalance({ sellToken: buyToken, buyToken: sellToken, from })
+
+      liquidityWasChecked = true
+    } catch (error) {
+      this.lastError = new Date()
+      liquidityWasChecked = false
+      this._handleError(sellToken, buyToken, error)
+    }
+
+    return liquidityWasChecked
+  }
+
+  async _checkEnoughBalance ({ sellToken, buyToken, from }) {
+    logger.debug('Checking enough balance for %s-%s in %s', sellToken, buyToken, from)
+    const auctionSellVolumePromise = this._dxInfoService.getSellVolume({
+      sellToken, buyToken })
+    const accountBalancesPromise = this._dxInfoService.getBalances({
+      address: from })
+    const auctionIndexPromise = this._dxInfoService.getAuctionIndex({
+      sellToken, buyToken })
+    const externalPricePromise = this._marketService.getPrice({
+      tokenA: sellToken, tokenB: buyToken })
+    const [ auctionSellVolume, accountBalances, auctionIndex, externalPrice ] =
+    await Promise.all([
+      auctionSellVolumePromise,
+      accountBalancesPromise,
+      auctionIndexPromise,
+      externalPricePromise
+    ])
+
+    const lastAvailableClosingPrice = await this._dxInfoService.getLastClosingPrices({
+      sellToken, buyToken, auctionIndex })
+    const { amount: buyTokenBalance } = accountBalances.find(({ token, amount }) => {
+      if (token === buyToken) {
+        return amount
+      }
+    })
+    const estimatedPrice = lastAvailableClosingPrice[lastAvailableClosingPrice.length - 1]
+      ? lastAvailableClosingPrice[lastAvailableClosingPrice.length - 1]
+      : externalPrice
+    const estimatedBuyVolume = auctionSellVolume.mul(numberUtil.toBigNumber(estimatedPrice))
+    logger.debug('Auction sell volume is %s %s and we have %s %s. With last available price %s %s/%s that should mean we need %s %s',
+      formatUtil.formatFromWei(auctionSellVolume), sellToken,
+      formatUtil.formatFromWei(buyTokenBalance), buyToken,
+      estimatedPrice, sellToken, buyToken,
+      formatUtil.formatFromWei(estimatedBuyVolume), buyToken)
+
+    if (estimatedBuyVolume.mul(1.10).greaterThan(buyTokenBalance)) {
```

Comment:
1.10 is cause you add a 10% of marging in case prices changes so we are not too tight?
That's a good idea, if that's the case, but we should use constants and give it a semantic name.

Reply:
Yes I was planning to add some margin so we aren't too close to the final price.

I'll do the semantic naming